File: JavaScript Front End/src/Messages/CommandMessages/GraphBuilding/AddTaskCommandMsg.js
const GraphBuildingCommandMsg = require('./GraphBuildingCommandMsg.js').GraphBuildingCommandMsg;

class AddTaskCommandMessage extends GraphBuildingCommandMsg
{
    constructor(taskName, taskTemporaryId)
    {
        super();
        this.jsonMessage["@type"] = this.jsonMessage["@type"].concat("AddTaskCommandMsg");
        this.jsonMessage["taskName"] = taskName;
        this.jsonMessage["taskTemporaryId"] = taskTemporaryId;
    }
}

module.exports = {AddTaskCommandMsg: AddTaskCommandMessage};

File: Java Back End/src/InfoSecCooker/Communication/Messages/CommandMessages/GraphBuilding/AddTaskCommandMsg.java
package InfoSecCooker.Communication.Messages.CommandMessages.GraphBuilding;

public class AddTaskCommandMsg extends GraphBuildingCommandMsg
{
    protected String taskName;
    protected long taskTemporaryId;

    public AddTaskCommandMsg(String taskName, long taskTemporaryId)
    {
        this.taskName = taskName;
        this.taskTemporaryId = taskTemporaryId;
    }

    public String getTaskName()
    {
        return taskName;
    }

    public long getTaskTemporaryId()
    {
        return taskTemporaryId;
    }

}
